### ISSUE TITLE
Word count for selected text

### ISSUE THREAD

**Issue by ladislas** [uranusjr/macdown#370]
Word count is very useful and I wonder if it could be possible to display the number of word of selected text.

Could be useful when different paragraphs need different number of words and/or characters :)

**Comment by uranusjr** [comment]
I never thought about this, but it would be a reasonable feature. The way how the counter is currently implemented doesn’t allow this, but this should be something to implement in the future. Thanks!

**Comment by ladislas** [comment]
you're welcome :)

**Comment by Jmuccigr** [comment]
I have an OS X service that does this. It shows up as "Statistics..." in my Services menu.

**Comment by ladislas** [comment]
can you share it? :)

**Comment by Jmuccigr** [comment]
Yeah...I was kind of hoping it was a standard one that was included with OS X, since I couldn't find it immediately upon searching. :-)

I think it's this one: http://www.devontechnologies.com/products/freeware.html

(My unwritten point though was that I would suggest developer time be spent on MacDown-specific things that aren't available already.)

**Comment by Fastidious** [comment]
I agree with @Jmuccigr. How useful will it be to have that feature, if, for example, my Markdown document contains lots of links, and other Markdown in it? Accuracy would be low, yes? Nevertheless, having it as a bonus--a plugin perhaps?--would not be that bad, but would prioritize MacDown Markdown features/bugs/etc first.

**Comment by uranusjr** [comment]
Due to the nature of Markdown, counting the source directly yields inaccurate results, as @Fastidious mentioned. The current counter takes this into account, and is actually counting words/characters *in the preview pane*, not the editor (see #52, which contains the original PR, and a pretty detailed discussion on this topic). That why I mentioned “how the counter is currently implemented doesn’t allow this”. To get an accurate word count, you need to go through the parser.

My point is, this *is* a Markdown-specific feature, albeit probably not an essential one, and will require support inside the core functionalities to work.

**Comment by Jmuccigr** [comment]
This confirms for me my original comment: since there's a service that handles this perfectly well (and in fact gives other stats as well), I'd prefer that developers spend their time on other things. I attach a screen shot of the service in action on some selected text in the MacDown preview pane. Any objections to having the docs give some discussion of this and include a link to this service?

![screenshot 2015-05-08 11 18 51](https://cloud.githubusercontent.com/assets/615115/7539317/284d84ec-f574-11e4-96b6-09f739d777c7.png)

**Comment by ladislas** [comment]
> This confirms for me my original comment: since there's a service that handles this perfectly well

@Jmuccigr yes you're right, but this requires to install new software/services, which might not always be possible and become a pain in the neck when you have to manage all the Macs of your company... Also it requires much more steps than just selecting the text in the file and having the answer right away.

I do a lot of work where the number of words/characters is tightly monitored and I just think this feature can be useful for people in the same case scenario.

I'm not saying it's the most important feature, neither that it should be done right away. But once every bug has been corrected it would be nice to have. As @uranusjr said, it can be done but needs underground work, so if I could be of any help, I'd be happy to get my hands on.

**Comment by konradedgar** [comment]
To add, [this article](http://hints.macworld.com/article.php?story=20120428125918199) on Mac OS X Hints, explains how to create new character counting service in Automator. For people willing to implement this solution - it's less than 5 minutes of work to add it.
